### ISSUE TITLE
Add torch.version.nccl

### ISSUE THREAD

**Issue by pietern** [pytorch/pytorch#31107]
## 🚀 Feature

Make NCCL version available at `torch.version.nccl`.

## Motivation

We have `torch.version.cuda` and should have a similar field for the NCCL version.

## Pitch

We should add it. The version is available as a compile time constant by looking at the header. Alternatively, we can pull the version dynamically through (`ncclGetVersion`)[https://docs.nvidia.com/deeplearning/sdk/nccl-developer-guide/docs/api/comms.html#ncclgetversion]. The latter has my preference, because it will always be correct, whereas the header approach leaves some ambiguity (e.g. you can use header of version X but load version Y).

## Additional context

Once added, this should be added to the version information scraper script as well.

**Comment by sheikmohdimran** [comment]
Don't we have `torch.cuda.nccl.version()` for this?

**Comment by pietern** [comment]
@sheikmohdimran You're right, we do! I was exploring what's in `torch.version` and didn't see `nccl` there. I still think we should stuff all version related information into `torch.version` for simplicity sake.
